platforms/cpuinfo.go

```diff
@@ -97,10 +125,21 @@ func getCPUVariant() string {
 
 	variant, err := getCPUInfo("Cpu architecture")
 	if err != nil {
-		log.L.WithError(err).Error("failure getting variant")
-		return ""
+		//If we can't find variant from /proc/cpuinfo, let's try uname -m.
```

Comment:
Does it make sense to keep the /proc/cpuinfo path? What is the benefit of using that instead of uname(2)?

Reply:
Current approach of using /proc/cpuinfo to get the cpu variant is tested working by users except this cross-platform situation listed in the PR. However the impact of changing to uname(2) from the previous approach can't be fully tested(we don't have CI environments to test different ARM archs, nor we lack local test environments to fully test that approach).      
Also ARM arch has a wide varieties of variants, and we have [edge case](https://github.com/containerd/containerd/blob/main/platforms/cpuinfo.go#L119) to handle. 
To maintain the stability of the project, I think patching the current solution is a better choice.

Reply:
Hmm, okay. Can you write down the reasoning on the code itself?

Also what would you need to test different ARM architectures? Isn't QEMU enough?

Reply:
I will add the reasoning to the code.

The piece of code will be executed on Linux platforms on different archs,  there are mainly three scenarios I can think of,  native x86/x86_64, native ARM with different variants, QEMU with a mix of host/guest environment.  For native ARM environment, we only have native ARM64 as CI platform, we will need at least ARM32 for a broad enough test coverage.

Reply:
I don't think we need to have mixes of host/guest environments for this particular function. All we need is /proc/cpuinfo and the result of uname(2) from various ARM environments.

You don't have to worry too much about CI right now. We can capture the values from ARM environments and hardcode them on this test.

Reply:
Sounds good about the lacking CI test envs. 

Any advice on getting the /proc/cpuinfo and result of uname(2) from varioous ARM environments? 

I only have access to ARM64 environments.